components/Menu.jsx

```diff
@@ -0,0 +1,101 @@
+import {
+  Article as ArticleIcon,
+  Home as HomeIcon,
+  Edit as EditIcon,
+  Class as ClassIcon,
+} from '@mui/icons-material';
+import {
+  Box,
+  Divider,
+  List,
+  ListItem,
+  ListItemButton,
+  ListItemIcon,
+  ListItemText,
+} from '@mui/material';
+import Link from 'next/link';
+import useStore from '../hooks/useStore';
+
+const mainItems = [
+  {
+    id: 0,
+    text: 'Home',
+    href: '/',
+    icon: <HomeIcon />,
+  },
+  {
+    id: 1,
+    text: 'All Blogposts',
+    href: '/',
+    icon: <ArticleIcon />,
+  },
+];
+
+const adminItems = [
+  {
+    id: 0,
+    text: 'Create Blogpost',
+    href: '/blog/create',
+    icon: <EditIcon />,
+  },
+  {
+    id: 1,
+    text: 'Create Category',
+    href: '/categories/create',
+    icon: <ClassIcon />,
+  },
+];
+
+export default function Menu({ toggleMenu }) {
```

Comment:
you are expecting a tunction here, but when you call Menu in Layout you don't hand over anything.